.github/workflows/typescript-test.yml

```diff
@@ -8,7 +8,7 @@ on:
   merge_group:
 
 jobs:
-  compile-and-test:
+  build-and-test:
```

Comment:
FYI I believe this job name is used in the required status checks, so we will have to update the required name after this merges

Reply:
Can you add that note into the PR description somewhere so it's a bit easier to remember?

Reply:
Ah okay, yes I can